logstash-core/lib/logstash/pipeline.rb

```diff
@@ -32,24 +32,34 @@ module LogStash; class Pipeline
     :started_at,
     :thread,
     :config_str,
+    :config_hash,
     :settings,
     :metric,
     :filter_queue_client,
-    :input_queue_client
+    :input_queue_client,
+    :xopts
```

Comment:
The basic idea was to provide extended options. Do you prefer `extended_options`? I dislike `options` because we already have settings, but `settings` unfortunately is for user configurable things and not rich objects ATM.